src/pineko/cli/fonll.py

```diff
@@ -0,0 +1,163 @@
+"""CLI entry point to FONLL."""
+import pathlib
+
+import click
+import rich
+
+from .. import configs, fonll, parser, theory_card
+from ._base import command
+
+config_setting = click.option(
+    "-c",
+    "--configs",
+    "cfg",
+    default=None,
+    type=click.Path(resolve_path=True, path_type=pathlib.Path),
+    help="Explicitly specify config file (it has to be a valid TOML file).",
+)
+
+
+class TheoryCardError(Exception):
+    """Raised when asked for FONLL theory cards with an original tcard as input that is not asking for FONLL."""
+
+    pass
+
+
+class InconsistentInputsError(Exception):
+    """Raised if the inputs are not consistent with FONLL."""
+
+    pass
+
+
+def load_cfg(name, grid):
+    """Path of the fktable in 'name' called 'grid' if it exists, else None."""
+    path = configs.configs["paths"]["fktables"] / name / grid
+    return path if path.exists() else None
+
+
+def get_list_grids_name(yaml_file):
+    """Return the list of the grids in the yaml file."""
+    yaml_content = parser._load_yaml(yaml_file)
+    # Turn the operands and the members into paths (and check all of them exist)
+    ret = []
+    for operand in yaml_content["operands"]:
+        for member in operand:
+            ret.append(f"{member}.{parser.EXT}")
+    return ret
+
+
+@command.command("combine_fonll")
+@click.argument("theoryID", type=int)
+@click.argument("dataset", type=str)
+@click.option("--FFNS3", type=int, help="theoryID containing the ffns3 fktable")
+@click.option("--FFN03", type=int, help="theoryID containing the ffn03 fktable")
+@click.option("--FFNS4", type=int, help="theoryID containing the ffns4 fktable")
+@click.option("--FFNS4til", type=int, help="theoryID containing the ffns4til fktable")
+@click.option("--FFNS4bar", type=int, help="theoryID containing the ffns4bar fktable")
+@click.option("--FFN04", type=int, help="theoryID containing the ffn04 fktable")
+@click.option("--FFNS5", type=int, help="theoryID containing the ffns5 fktable")
+@click.option("--FFNS5til", type=int, help="theoryID containing the ffns5til fktable")
+@click.option("--FFNS5bar", type=int, help="theoryID containing the ffns5bar fktable")
```

Comment:
Instead of a thousand options, couldn't we just use a list of integers with some encoding?

Reply:
Maybe yes but they are all options because they are all optional. I believe that following your suggestion we may lose that flexibility or at least in clarity.

Reply:
Actually, is this really the case? I believe you can choose to not have a flavor, or some part of it, but not all the options are independent.

We may have a scheme to always interpret the list.

Reply:
From the code point of view you have full flexibility. Of course from the physics point of view there are some choices that just do not make sense. The code would not fail though